pilot/pkg/networking/core/listener_builder.go

```diff
@@ -188,6 +199,9 @@ func (lb *ListenerBuilder) patchListeners() {
 	lb.virtualInboundListener = lb.patchOneListener(lb.virtualInboundListener, networking.EnvoyFilter_SIDECAR_INBOUND)
 	lb.httpProxyListener = lb.patchOneListener(lb.httpProxyListener, networking.EnvoyFilter_SIDECAR_OUTBOUND)
 	lb.inboundListeners = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_SIDECAR_INBOUND, lb.envoyFilterWrapper, lb.inboundListeners, false)
+	if lb.node.Type == model.Waypoint {
```

Comment:
Do we need to apply this to the entire listener? I am worried about correctness and performance.

For performance, are we doing more work by traversing the entire listener instead of looking at each filter chain in isolation?

For correctness. We set `lb.envoyFilterWrapper` to "All envoyfilter selected by any relevant service", but then we apply these to _all_ filter chains. However, I may have `FC-svc-1` and `FC-svc-2` and I need to apply the appropriate subset of filters